cartography/intel/azure/util/credentials.py

```diff
@@ -0,0 +1,173 @@
+import logging
+from datetime import datetime
+from datetime import timedelta
+from typing import Any
+from typing import Optional
+
+import adal
+import requests
+from azure.common.credentials import get_azure_cli_credentials
+from azure.common.credentials import get_cli_profile
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.core.exceptions import HttpResponseError
+from msrestazure.azure_active_directory import AADTokenCredentials
+
+
+logger = logging.getLogger(__name__)
+AUTHORITY_HOST_URI = 'https://login.microsoftonline.com'
+
+
+class Authenticator:
+
+    def authenticate_cli(self) -> Any:
```

Comment:
If I read this correctly, the return type here should be `Credential` and not `Any`

Reply:
updated in latest commit